src/test.ts

```diff
@@ -14,22 +14,72 @@ function normalizeLineEndings(text: string): string {
   return text.replace(/\r\n?/g, "\n");
 }
 
-function compareToBaselines() {
-  for (const file of fs.readdirSync(baselineFolder)) {
+function compareToBaselines(baselineFolder: URL, outputFolder: URL) {
+  let baselineFiles: string[] = [];
+  try {
+    baselineFiles = fs.readdirSync(baselineFolder);
+  } catch {
+    // do nothing
+  }
+
+  let outputFiles: string[] = [];
+  try {
+    outputFiles = fs.readdirSync(outputFolder);
+  } catch {
+    // do nothing
```

Comment:
Are we actually expecting an error here? This looks like it could potentially silence a real failure 🤔

Reply:
the failures that matter are mismatched files. If either the input or output dir don't exist, then the file mismatch will be reported when we traverse whichever side did exist and find the file on the other side is missing.

This actually catches *more* failures as we previously only checked what was in `baseline` and ignored new things in `generated`. This now enumerates all distinct entries from both so we will report on excess baselines or excess outputs.

Reply:
Yeah, but we call stat before entering directory, so not sure this one matters.

Reply:
We call stat, but we still traverse the directory if the stat only succeeded on one side or the other, so the `readddir` can still potentially fail.